pkg/storage/swift/swift.go

```diff
@@ -116,6 +116,7 @@ func (d *driver) ConfigEnv() (envs []corev1.EnvVar, err error) {
 	d.Config.Domain = replaceEmpty(d.Config.Domain, cfg.Storage.Swift.Domain)
 	d.Config.DomainID = replaceEmpty(d.Config.DomainID, cfg.Storage.Swift.DomainID)
 	d.Config.RegionName = replaceEmpty(d.Config.RegionName, cfg.Storage.Swift.RegionName)
+	d.Config.AuthVersion = replaceEmpty(d.Config.AuthVersion, "3")
```

Comment:
Do we only support v3 for auto-configuration?

Reply:
Technically speaking for a long time v3 is the only version available: v1 never existed publicly, and v2 was deprecated about 6 years ago, and completely removed from OpenStack at the end of 2017.
https://blueprints.launchpad.net/keystone/+spec/deprecate-v2-api
https://docs.openstack.org/keystone/stein/contributor/http-api.html
Therefore all existing OSP releases use v3 API and don't allow to switch to v2.
For this reason, I believe that making v3 the default API version is absolutely justified.